### PR TITLE
Cleanup sync v1 migration

### PR DESCRIPTION
**Pull request by AlexeyBarabash**
Resolves https://github.com/brave/brave-browser/issues/33113

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by fallaciousreasoning — APPROVED** [review]
Wow, awesome!

File: chromium_src/components/infobars/core/infobar_delegate.h [hunk 1]
```diff
@@ -14,14 +14,16 @@
...
-  SYNC_V2_MIGRATE_INFOBAR_DELEGATE = 503,                                      \
-  ANDROID_SYSTEM_SYNC_DISABLED_INFOBAR = 504, SYNC_CANNOT_RUN_INFOBAR = 505,   \
+  WAYBACK_MACHINE_INFOBAR_DELEGATE = 502, SYNC_CANNOT_RUN_INFOBAR = 505,       \
   WEB_DISCOVERY_INFOBAR_DELEGATE = 506,                                        \
   BRAVE_SYNC_ACCOUNT_DELETED_INFOBAR = 507,                                    \
   BRAVE_REQUEST_OTR_INFOBAR_DELEGATE = 508, BRAVE_IPFS_INFOBAR_DELEGATE = 509, \
   DEV_CHANNEL_DEPRECATION_INFOBAR_DELEGATE = 510,
 
+// Deprecated:
+// SYNC_V2_MIGRATE_INFOBAR_DELEGATE = 503
```
Comment (mkarolin) [inline chromium_src/components/infobars/core/infobar_delegate.h:24]: Should this value also be commented out in brave/android/java/org/chromium/chrome/browser/infobar/BraveInfoBarIdentifier.java?

**Review by mkarolin — APPROVED** [review]
`strings`++
`chromium_src`/`patches`++

**Review by SergeyZhukovsky — APPROVED** [review]
android tests part++

File: chromium_src/components/infobars/core/infobar_delegate.h (hunk 1 above)
Comment (AlexeyBarabash) [reply chromium_src/components/infobars/core/infobar_delegate.h:24]: Yes, it should be commented out there also.
Here is that place:
https://github.com/brave/brave-core/pull/20906/files#diff-904cb9b3f638eaabd683bab4bc469c0ac20dcce6f6a30afca30b83be10f660c6R26

**Review by simonhong — APPROVED** [review]
👍🏼

**Review by cdesouza-chromium — APPROVED** [review]

**Comment by AlexeyBarabash** [comment]
Re-based just in case to minimize the risk of conflicts

**Pull request merged**